### DISCUSSION TITLE
How to implement variability of non decision time in PYDDM using DDM model?

### DISCUSSION THREAD

**Discussion by DianaY-J** [mwshinn/PyDDM#99] [Ask for help] [answered]
Hi omnipotent Github friends,

Recently my colleague and I are trying to fit DDM model to my data, and I want to use variability of non decision time to solve the prematures responses. We think it would be more elegant way than using cut off to get rid of those. We want to use it on simple ddm model without necessarily using gddm model. We thought there must be people trying those before, but we can't find anything in github or in PYDDM cookbook. Does anyone have any idea about how to implement it? 

Thank you so much in advance!

**Comment by mwshinn** [comment]
Hi Diana!  If you use the new interface (which I recommend, it is a lot easier!) see:

https://pyddm.readthedocs.io/en/latest/cookbook/modelcomponents.html#non-decision-variability

**Reply by DianaY-J** [reply]
Hey Max, thank you so much! I saw this before, but currently we would prefer a ddm model over a gddm model for the simplicity of our data, do you have any idea how to incorporate that into a normal ddm model? Or do we have to use a gddm model to be able to implement variability of non decision time?

**Reply by mwshinn** [reply, marked as answer]
A DDM model with reaction time variability is by definition a GDDM model!  All DDMs are also GDDMs (but not vice versa), and PyDDM doesn't distinguish between the two.  So if you've built a model in PyDDM, you've already built a GDDM.

**Reply by DianaY-J** [reply]
Hi Max, thank you so much! Then is it possible to also implement other features such as defining a range for drift and boundary, defining a constant for noise using Fittable in gddm? Like the only thing i change from ddm to gddm is overlay?

**Reply by mwshinn** [reply]
Yes, there is a new syntax, see the quickstart guide: https://pyddm.readthedocs.io/en/latest/quickstart.html

**Reply by DianaY-J** [reply]
awesome, i'll try that. Thank you so much!!
